easybuild/tools/modules.py

```diff
@@ -1340,6 +1340,8 @@ def avail_modules_tools():
     # filter out legacy Modules class
     if 'Modules' in class_dict:
         del class_dict['Modules']
+    if 'NoModulesTool' in class_dict:
```

Comment:
maybe add a comment above this line to explain why: `# NoModulesTool should never be used deliberately, so remove it from the list of available module tools`